staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/model/adaptor.go

```diff
@@ -143,10 +212,106 @@ func (s *Structural) XListType() string {
 	return *s.Structural.XListType
 }
 
+func (s *Structural) XMapType() string {
+	if s.Structural.XMapType == nil {
+		return ""
+	}
+	return *s.Structural.XMapType
+}
+
 func (s *Structural) XListMapKeys() []string {
 	return s.Structural.XListMapKeys
 }
 
+func (s *Structural) AllOf() []common.Schema {
+	var res []common.Schema
+	for _, subSchema := range s.Structural.ValueValidation.AllOf {
+		subSchema := subSchema
+		res = append(res, nestedValueValidationToStructural(&subSchema))
```

Comment:
Could you comment on why a conversion is preferred over 
```go
res = append(res, &Structural{Structural: &subSchema})
```

Reply:
`subSchema` is of type `NestedValueValidation`, not `Structural`

Reply:
I see. It is a small surprise to see `Structural` did not follow the recursive definition pattern. However, a conversion is better avoided still (see below).